sharding/handler/utils/log_parser.py

```diff
@@ -43,18 +51,18 @@ def _set_topic_value(self, *, topics, log):
             val = self._parse_value(val_type=topic[1], val=log['topics'][i + 1])
             setattr(self, topic[0], val)
 
-    def _set_data_value(self, *, data, log):
+    def _set_data_value(self, *, data: List[Tuple[str, Any]], log: Dict[str, Any]) -> None:
         data_bytes = decode_hex(log['data'])
         if len(data) * 32 != len(data_bytes):
             raise LogParsingError(
                 "Error parsing log data, expect"
                 "{} data but get {}.".format(len(data), len(data_bytes))
             )
-        for (i, data) in enumerate(data):
-            val = self._parse_value(val_type=data[1], val=data_bytes[i * 32: (i + 1) * 32])
-            setattr(self, data[0], val)
+        for (i, (name, type_)) in enumerate(data):
```

Comment:
👍

Reply:
That one was interesting to me: Actually `mypy` complained about that line not because `data` was reassigned but because it was reassigned with a new type.